packages/core/src/browser/save-resource-service.ts

```diff
@@ -14,15 +14,153 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { inject, injectable } from 'inversify';
-import { MessageService, UNTITLED_SCHEME, URI } from '../common';
+import type { ApplicationShell } from './shell';
+import { injectable } from 'inversify';
+import { UNTITLED_SCHEME, URI, Disposable, DisposableCollection } from '../common';
 import { Navigatable, NavigatableWidget } from './navigatable-types';
-import { Saveable, SaveableSource, SaveOptions } from './saveable';
+import { AutoSaveMode, Saveable, SaveableSource, SaveOptions, SaveReason } from './saveable';
 import { Widget } from './widgets';
+import { FrontendApplicationContribution } from './frontend-application-contribution';
+import { FrontendApplication } from './frontend-application';
+import throttle = require('lodash.throttle');
 
 @injectable()
-export class SaveResourceService {
-    @inject(MessageService) protected readonly messageService: MessageService;
+export class SaveResourceService implements FrontendApplicationContribution {
```

Comment:
This service saves `Saveable` instances, not resources. If we _do_ have  breaking changes anyway, I would rename it.